### DISCUSSION TITLE
Access policy probabililties in engine

### DISCUSSION THREAD

**Discussion by SimonBenhamou** [LeelaChessZero/lc0#1906] [Q&A]
Hello,

I am trying to access the maia networks policy (move probabilities given a position).

In the python bindings I can get the move probabilities with the p_softmax function, which makes this quite easy.

However, I don't find this information when using the UCI engine... is this exposed in any way ?

Thanks!

**Comment by mooskagh** [comment]
Hi SimonBenhamou,

There is no convenient way to access raw P values through UCI.  
The best possible way to do it without modifying the code is to run Lc0 with the `--verbose-move-stats` parameter. In that case, after the search is done, Lc0 will output some stats about the position currently being searched through uci `info string` lines, and that includes P.

**Comment by SimonBenhamou** [comment]
@mooskagh verbose-move-stats does answer my request, thanks a lot. 

In order to expose it through the python UCI API, I did also find a way to do it, see the following python-chess issue I submitted: https://github.com/niklasf/python-chess/issues/1026
